notebooks/source/bayesian_hierarchical_linear_regression.ipynb

```diff
@@ -0,0 +1,677 @@
+    "def model(PatientID, Weeks, FVC_obs=None):\n",
+    "    mu_a = numpyro.sample(\"mu_a\", dist.Normal(0., 100.))\n",
+    "    sigma_a = numpyro.sample(\"sigma_a\", dist.HalfNormal(100.))\n",
+    "    mu_b = numpyro.sample(\"mu_b\", dist.Normal(0., 100.))\n",
+    "    sigma_b = numpyro.sample(\"sigma_b\", dist.HalfNormal(100.))\n",
+    "    \n",
+    "    unique_patient_IDs = np.unique(PatientID)\n",
+    "    n_patients = len(unique_patient_IDs)\n",
+    "    \n",
+    "    a = numpyro.sample(\"a\", dist.Normal(mu_a, sigma_a).expand([n_patients]))\n",
+    "    b = numpyro.sample(\"b\", dist.Normal(mu_b, sigma_b).expand([n_patients]))\n",
+    "    \n",
+    "    sigma = numpyro.sample(\"sigma\", dist.HalfNormal(100.))\n",
+    "    \n",
+    "    FVC_est = a[PatientID] + b[PatientID] * Weeks\n",
+    "    \n",
+    "    with numpyro.plate(\"data\", len(PatientID)):\n",
+    "        numpyro.sample(\"obs\", dist.Normal(FVC_est, sigma), obs=FVC_obs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That's all for modelling!\n",
+    "\n",
+    "## 3. Fitting the model\n",
+    "A great achievement of Probabilistic Programming Languages such as Numpyro is to decouple model specification and inference. After specifying my generative model, with priors, condition statements and data likelihood, I can leave the hard work to Numpyro's inference engine. \n",
+    "\n",
+    "Calling it requires just a few lines. Before we do it, let's add a numerical Patient ID:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.preprocessing import LabelEncoder\n",
+    "\n",
+    "le = LabelEncoder()\n",
+    "train['PatientID'] = le.fit_transform(train['Patient'].values)\n",
+    "\n",
+    "FVC_obs = train['FVC'].values\n",
+    "Weeks = train['Weeks'].values\n",
+    "PatientID = train['PatientID'].values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, calling Numpyro's inference engine:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "sample: 100%|██████████| 4000/4000 [00:28<00:00, 139.79it/s, 63 steps of size 1.08e-01. acc. prob=0.88] \n"
+     ]
+    }
+   ],
+   "source": [
+    "nuts_kernel = NUTS(model)\n",
+    "\n",
+    "mcmc = MCMC(nuts_kernel, num_samples=2000, num_warmup=2000)\n",
+    "rng_key = random.PRNGKey(0)\n",
+    "mcmc.run(rng_key, PatientID, Weeks, FVC_obs=FVC_obs)\n",
+    "\n",
+    "posterior_samples = mcmc.get_samples()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Checking the model\n",
+    "### 4.1. Inspecting the learned parameters\n",
+    "First, let's inspect the parameters learned. To do that, I will use Arviz, which perfectly integrates with Numpyro:"
```

Comment:
nit: Arviz -> ArviZ. Also, could you add a link to [ArviZ](https://arviz-devs.github.io/arviz/) so readers can know how to install it and get an overview about this package?